parser/parser.go

```diff
@@ -200,7 +203,18 @@ func (p *Parser) expectPeek(t token.Type) bool {
 // creates an error and adds it to the parser errors list
 func (p *Parser) peekError(t token.Type) {
 	msg := fmt.Sprintf("unexpected token: %q (expected: %q) at line: %d", p.peekToken.Type, t, p.lexer.RowNum)
+	p.errors = append(p.errors, msg)
+}
+
+func (p *Parser) checkIfOverridesBuiltin() {
+	if _, ok := builtins[p.curToken.Literal]; ok {
+		msg := fmt.Sprintf("cannot override built-in function: %q at line: %d", p.curToken.Literal, p.curToken.LineNumber)
+		p.errors = append(p.errors, msg)
+	}
+}
 
+func (p *Parser) semiError() {
```

Comment:
semi means half xd maybe missingSemicolonError or sth similar would be better?

Reply:
well, go ahead then 😄